core/db/migrate/20150113002122_create_spree_promotion_codes.rb

```diff
@@ -0,0 +1,13 @@
+class CreateSpreePromotionCodes < ActiveRecord::Migration
+  def change
+    create_table :spree_promotion_codes do |t|
+      t.references :promotion, index: true
+      t.string :value, unique: true
+      t.datetime :starts_at
+      t.datetime :expires_at
+      t.integer :usage_limit
+
+      t.timestamps
```

Comment:
promotion_id was already indexed (`t.references :promotion, index: true`).
Added the indexes for `value`, `starts_at` and `expires_at` in the lines below.